### ISSUE TITLE
Thanos sidecar container is restarting due to OOMKilled

### ISSUE THREAD

**Issue by SanthoshiMekala** [thanos-io/thanos#5963]
Hi Team,

We have thanos as a sidecar in Prometheus. We have some highest cardinality metrics. For one of the metrics, we have above 20 lacs timeseries data. When querying for count of that metric, we are getting "context deadline exceeded" error in thanos query and the thanos-sidecar is getting restarted due to OOMKilled. The resources config for the thanos-sidecar is as below.
  Limits:
      cpu:     200m
      memory:  2Gi
    Requests:
      cpu:     65m
      memory:  512Mi

Thanos Version: v0.24.0
Prometheus Version: v2.34.0

Thanks in advance!

**Comment by SanthoshiMekala** [comment]
Hi @yeya24 I see the CPU and memory consumption very low. Still, do we need to increase the resources?
![image](https://user-images.githubusercontent.com/39329333/207508371-a0a1d339-fcbb-46fc-a70b-48d1bd8c6e06.png)

**Comment by yeya24** [comment]
I think 2GB is enough for sidecar. But since you mentioned it is OOM killed, do you happen to have any profile about sidecar?
It is OOM killed when a query requests a lot of series or when did that happen?

**Comment by SanthoshiMekala** [comment]
Yes @yeya24 whenever we are trying to query for a metric with huge data, its getting OOMKilled. But, the usage in kibana is not showing spikes. Do we need to check anywhere? Or do we have any field to set the heap for thanos like that?

**Comment by yeya24** [comment]
@SanthoshiMekala No spikes probably only means your monitoring system didn't catch the OOM memory spike but it happened anyway. I think you could try making GOGC more aggresive.

Anyway, if you are querying huge data then I think 2GB is not enough. Try allocating more or reducing your cardinalities.
